R/load_package.r

```diff
@@ -1,15 +1,89 @@
 #' Load a package and install it if necessary
 #' 
 #' @name load_package
-#' @param name Name of package
+#' @param name character. Name of package.
+#' @param verbose logical. Whether or not to announce each installation.
 #' @examples
 #' \dontrun{
-#' load_package('glmnet')
+#' load_package("glmnet")
+#' load_package("robertzk/Ramd")
+#' load_package("robertzk/Ramd@@v0.3")   # Can load from versions
+#' load_package("robertzk/Ramd@@0.3")
+#' load_package("robertzk/Ramd@@fbe1aa0e36df289b27881d077635352e6debdbc1")  # Can load from refs
+#' load_package(list("FeiYeYe/xgboost", subdir = "R-package"))  # Can load from subdirectories
 #' }
-load_package <- function(name) {
-  if (!require(name, character.only = TRUE)) {
-    install.packages(name, dependencies = TRUE)
-    if (!require(name, character.only = TRUE))
-      stop(paste('Package', name, 'not found'))
+load_package <- function(name, verbose = FALSE) {
+  if (is.list(name)) {
```

Comment:
You can replace this whole section with:

```r
metadata <- name[-1]
name <- name[[1]]
```

and then below replace

```r
if (!is.null(metadata)) {
  do.call(devtools::install_github, c(name, metadata))
}
```

with

```r
if (length(metadata)) {
  do.call(devtools::install_github, c(list(name), metadata))
}
```